### PR TITLE
Add statistics for evictions and cache size. Simplify rotation logic

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: src/HitStatistics.js [hunk 1]
```diff
@@ -3,34 +3,59 @@ import { getTimestamp } from './utils/dateUtils.js'
...
+    this.collectionStart = new Date()
+    this.currentTimeStamp = getTimestamp(this.collectionStart)
+
     this.records = globalStatisticsRecord || new HitStatisticsRecord()
     this.records.initForCache(this.cacheId, this.currentTimeStamp)
   }
 
+  get currentRecord() {
+    // safety net
+    /* c8 ignore next 9 */
```
Comment (kibertoad) [inline src/HitStatistics.js:18]: I can't reproduce this, but this place crashed in production :-/

File: src/HitStatistics.js [hunk 2]
```diff
@@ -39,15 +64,7 @@ export class HitStatistics {
 
   archiveIfNeeded() {
     if (this.hoursPassed() >= this.statisticTtlInHours) {
-      const now = new Date()
```
Comment (kibertoad) [inline src/HitStatistics.js:42]: we don't need all this logic, `initForCache` rewrites entire entry to an empty record anyway

**Review by gurgunday — COMMENTED** [review]
I was currently reviewing :)

the code looks good to me

**Comment by kibertoad** [comment]
@gurgunday ah, sorry, I should have pinged you. needed to release these changes, as they were blocking me already, but if anything still comes up, would be happy to fix it in follow-up PR!

**Pull request merged**